### PR TITLE
Fail fast on invalid image

### PR DESCRIPTION
**Pull request by afrittoli**
# Changes

The kubernets pod treats an invallid image failures as potentially ephemeral errors, because even if the format of the image reference is not syntactically correct, users may update image without recreating the Pod.

Tekton, however, uses Pod to provide workloads that run to completion, and users are not allowed to change the specification of steps during execution.

This commits changes the handling of the InvalidImageName pod reason, so that the TaskRun is marked as failed and the Pod deleted.

Fixes: #6105

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
The Pod reason InvalidImageName is treated now as a permanent issue, so that TaskRuns that include a step with an invalid image reference are failed immediately and the corresponding Pod is deleted.
```

/kind bug

### CODE REVIEW

File: pkg/reconciler/taskrun/taskrun_test.go [hunk 1]
```diff
@@ -2314,65 +2314,36 @@ status:
 	}
 }
 
-func TestReconcilePodFailuresStepImagePullFailed(t *testing.T) {
```
Comment (afrittoli) [inline pkg/reconciler/taskrun/taskrun_test.go:2317]: I turned the next test into a matrix test, and the two cases in here are included in the next test now.

**Comment by afrittoli** [comment]
Example `TaskRun` with the fix in:


```
➜ tkn tr describe step-script-svs9s
Name:              step-script-svs9s
Namespace:         default
Service Account:   default
Timeout:           1h0m0s
Labels:
 app.kubernetes.io/managed-by=tekton-pipelines

🌡️  Status

STARTED         DURATION    STATUS
3 minutes ago   42s         Failed(TaskRunImagePullFailed)

Message

The step "bash" in TaskRun "step-script-svs9s" failed to pull the image "". The pod errored with the message: "Failed to apply default image tag "ubuntu@1234:v45@12fres": couldn't parse image reference "ubuntu@1234:v45@12fres": invalid reference format."

🦶 Steps

 NAME                          STATUS
 ∙ bash                        TaskRunImagePullFailed
```

**Review by vdemeester — APPROVED** [review]

**Comment by afrittoli** [comment]
@JeromeJu @vdemeester can I get an lgtm?

**Comment by JeromeJu** [comment]
/lgtm
sorry Andrea, wanted to have more eyes on this earlier. This looks good to me1

**Pull request merged**